src/Microsoft.AspNetCore.Mvc.Razor.Host/ModelExpressionPass.cs

```diff
@@ -55,10 +58,35 @@ public override void VisitSetTagHelperProperty(SetTagHelperPropertyIRNode node)
                         });
                     }
                     else
-                    {
```

Comment:
Unit testing found a bug here. We were just copying all of the nodes into the new expression, but that isn't quite right. We also uncovered a bug in Razor. Once this is addressed, won't need to handle "HTML" here (note the scare quotes).